src/SdkCommon/ClientRuntime/ClientRuntime/ServiceClient.cs

```diff
@@ -472,8 +482,8 @@ public bool SetUserAgent(string productName, string version)
             if (!_disposed && HttpClient != null)
             {
                 MergeUserAgentInfo(DefaultUserAgentInfoList);
-                string cleanedProductName = CleanUserAgentInfoEntry(productName);                
-                HttpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(cleanedProductName, version));
+                string cleanedProductName = CleanUserAgentInfoEntry(productName);
+                MergeUserAgentInfo(new ProductInfoHeaderValue(cleanedProductName, version));
```

Comment:
What we really need is the deDuping of UserAgent collection.
Yes this unblocks you for the nullRef, but this has perf. impact and I would prefer a deDuping of UserAgent collection than what is implemented in your PR.
So if you can update your PR with 1 time deDuping of UserAgent collection that will be great, else we can do that.

Reply:
I've made the requested changes.

Frankly, this method for deDuping UserAgent isn't enough for my use case.  I'd like to use the constructor that allows me to pass in an HttpClient.  However, I want to use a Static HttpClient ... and this code is not threadsafe.  I believe a better option would be to not add UserAgent headers at all, if the consumer has provided their own HttpClient.  What are your thoughts?